source/coverage.cmake

```diff
@@ -0,0 +1,3 @@
+if(YOTTA_CFG_DEBUG_OPTIONS_COVERAGE_MODULES_SOCKETS)
+  set( CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -fprofile-arcs -ftest-coverage" )
```

Comment:
better to use `set_target_properties` than `set(CMAKE_CXX_FLAGS)`, this could potentially be inherited by other directories.